src/utils/beneficial.owner.statements.ts

```diff
@@ -6,33 +6,40 @@ import { logger } from "../utils/logger";
 import { ApplicationData } from "../model";
 import { checkBOsDetailsEntered, checkMOsDetailsEntered, getApplicationData, setExtraData } from "../utils/application.data";
 import { BeneficialOwnersStatementType, BeneficialOwnerStatementKey } from "../model/beneficial.owner.statement.model";
-import { EntityNumberKey } from "../model/data.types.model";
+import { EntityNameKey, EntityNumberKey } from "../model/data.types.model";
 import { saveAndContinue } from "../utils/save.and.continue";
+import { isActiveFeature } from "./feature.flag";
+import { containsTrustData, getTrustArray } from "./trusts";
 
 export const getBeneficialOwnerStatements = (req: Request, res: Response, next: NextFunction, registrationFlag: boolean, noChangeBackLink?: string) => {
   try {
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
 
-    let BACK_LINK: string;
+    let backLinkUrl: string;
     let noChangeFlag: boolean = false;
     let templateName: string;
-
+    const appData = getApplicationData(req.session);
     if (noChangeBackLink){
-      BACK_LINK = noChangeBackLink;
+      backLinkUrl = noChangeBackLink;
       templateName = config.UPDATE_NO_CHANGE_BENEFICIAL_OWNER_STATEMENTS_PAGE;
       noChangeFlag = true;
     } else {
-      BACK_LINK = registrationFlag ? config.ENTITY_URL : config.OVERSEAS_ENTITY_REVIEW_URL;
+      if (isActiveFeature(config.FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION)){
+        backLinkUrl = containsTrustData(getTrustArray(appData))
+          ? config.UPDATE_TRUSTS_ASSOCIATED_WITH_THE_OVERSEAS_ENTITY_URL
```

Comment:
should there be a trusts flag check here as well 🤢

Reply:
or can we assume that if update trusts feature flags is off we'll never have trust data?

Reply:
Not really as if they dont have any bo's with trust NOC's then they would've never went through the trust flow, so no trusts @AndreKramer7654